clash-lib/src/Clash/Core/Type.hs

```diff
@@ -139,12 +140,12 @@ varAttrs (Id _ _ ty _) =
 
 -- | Types in CoreHW: function and polymorphic types
 data Type
-  = VarTy    !TyVar             -- ^ Type variable
-  | ConstTy  !ConstTy           -- ^ Type constant
-  | ForAllTy !TyVar !Type       -- ^ Polymorphic Type
-  | AppTy    !Type !Type        -- ^ Type Application
-  | LitTy    !LitTy             -- ^ Type literal
-  | AnnType  [Attr'] !Type      -- ^ Annotated type, see Clash.Annotations.SynthesisAttributes
+  = VarTy    !TyVar              -- ^ Type variable
+  | ConstTy  !ConstTy            -- ^ Type constant
+  | ForAllTy !TyVar !Type        -- ^ Polymorphic Type
+  | AppTy    !Type !Type         -- ^ Type Application
+  | LitTy    !LitTy              -- ^ Type literal
+  | AnnType  [Attr String] !Type -- ^ Annotated type, see Clash.Annotations.SynthesisAttributes
```

Comment:
I used it for the user facing types because:

 * I didn't want users to have to deal with `OverloadedStrings`
 * I was afraid `Text` wouldn't survive Clash normilazation

But yeah, there's no reason to not use `Text` in `Clash.Core.Type`. I'll change it!